### ISSUE TITLE
Put id attributes on TOC entries so they can be linked to

### ISSUE THREAD

**Issue by MuntashirAkon** [jgm/pandoc#7907]
**Describe your proposed improvement and the problem it solves.**
Pandoc has an option to generate table of contents. It also nicely links the section titles making it a bit easy to navigate around the generated HTML file. However, there is no way of going back to the table of content from the section that the reader has navigated to.

Here's a link to a documentation generated by `make4ht` as an example: https://muntashirakon.github.io/AppManager/en/

As you can see in the example,
1. Section numbers do not have hyperlinks, only the titles do, thus, making it easier to distinguish the titles from the section number
2. It is possible go back to the table of contents via the hyperlink in the section title.

**Describe alternatives you've considered.**
I can continue to use `make4ht` but it has other issues. Other option is to write a script to emulate that myself.

**Comment by jgm** [comment]
I think this is best handled with customization on top of what pandoc does.
For example, on <https://pandoc.org> we use a small Lua script to insert the anchor links that appear on hover:

```lua
-- Adds anchor links to headings with IDs.
function Header (h)
  if h.identifier ~= '' then
    -- an empty link to this header
    local anchor_link = pandoc.Link(
      {},                  -- content
      '#' .. h.identifier, -- href
      '',                  -- title
      {class = 'anchor', ['aria-hidden'] = 'true'} -- attributes
    )
    h.content:insert(1, anchor_link)
    return h
  end
end
```

You could modify this to make it link to the TOC.  Everyone is going to have different preferences for this kind of thing, and making the section headings links by default is not what everyone wants.  (Actually, pandoc used to do this long ago but we stopped.)

**Comment by tarleb** [comment]
I think these links will come in handy for this: 

- <https://stackoverflow.com/q/69891283/>
- <https://groups.google.com/g/pandoc-discuss/c/1xIAG684yrM/m/SANzToYFFwAJ>

**Comment by MuntashirAkon** [comment]
> Hence the second link, which shows how you can create your own toc with a Lua filter.

Ah, thanks!

**Comment by jgm** [comment]
Maybe we should modify the HTML writer so that every TOC entry gets its own id attribute, generated in some predictable way based on the id it links to?  That would facilitate this.

**Comment by jgm** [comment]
I'll reopen this issue and rebrand it accordingly.

**Comment by damon-sava-stanley** [comment]
Howdy. Looked at this quickly. Apologies if this is a bother, but would the idea just be to add an id to the `headerLink` below (e.g. `"#toc-"` + the identifier it links to). This would be picked up by a couple other writers, but that might be desired. Anyways, if this is what is wanted, I'd be happy to give it a go.

https://github.com/jgm/pandoc/blob/97d83e383ad8889ea2003ce74f728d51e4bbb248/src/Text/Pandoc/Writers/Shared.hs#L434-L451

**Comment by jgm** [comment]
Yes, that's the basic idea.  Might need a little fine-tuning when we see what it looks like.

**Comment by damon-sava-stanley** [comment]
Submitted a pull request, which just adds "toc-" + header-id to the TOC links. If we wanted to do anything else, e.g. gate it behind a writer option, its open to revision

**Comment by jgm** [comment]
Closed by #7913.
